lib/ansible/plugins/action/synchronize.py

```diff
@@ -200,14 +203,11 @@ def run(self, tmp=None, task_vars=dict()):
                     user = task_vars.get('ansible_ssh_user') or self._play_context.remote_user
 
             # use the mode to define src and dest's url
-            if self._task.args.get('mode', 'push') == 'pull':
-                # src is a remote path: <user>@<host>, dest is a local path
-                src = self._process_remote(src_host, src, user)
-                dest = self._process_origin(dest_host, dest, user)
-            else:
-                # src is a local path, dest is a remote path: <user>@<host>
-                src = self._process_origin(src_host, src, user)
-                dest = self._process_remote(dest_host, dest, user)
+            set_mode = lambda host, var: \
+                          self._process_origin(host, var, user) if host == '127.0.0.1' \
+                     else self._process_remote(host, var, user)
+            src = set_mode(src_host, src)
+            dest = set_mode(dest_host, dest)
         else:
```

Comment:
Ah.. I see that push and pull are used to switch src_host and dest_host earlier in the file.  So is this portion an actual fix or just refactoring?